lib/modules/manager/helmfile/readme.md

```diff
@@ -13,3 +13,40 @@ The `helmfile` manager defines this default registryAlias:
 If your Helm charts make use of repository aliases then you will need to configure an `registryAliases` object in your config to tell Renovate where to look for them. Be aware that alias values must be properly formatted URIs.
 
 If you need to change the versioning format, read the [versioning](https://docs.renovatebot.com/modules/versioning/) documentation to learn more.
+
+### Private repositories and registries
+
+To use private sources of Helm charts, you must set the password and username you use to authenticate to the private source.
+For this you use a custom `hostRules` array.
+
+#### OCI registries
+
+```json5
+{
+  hostRules: [
+    {
+      // global login
+      matchHost: 'ghcr.io',
+      hostType: 'docker',
+      username: '<some-username>',
+      password: '<some-password>',
+    },
+    {
+      // for repository string oci://ghci.io/user/oci-helm-test
+      matchHost: 'https://ghci.io/user/oci-helm-test',
```

Comment:
```suggestion
      matchHost: 'https://ghci.io',
```
doesn't work with docker datasource , because the full url looks different and it uses the login realm url to find the host rules.

checkout this test

https://github.com/renovatebot/renovate/blob/43d3e343b769f3198eec98b2937fdb764fa5cd45/lib/modules/datasource/docker/index.spec.ts#LL2013C3-L2013C3